File: packages/amplify_core/lib/src/types/datastore/hub/outbox_status_event.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import 'package:amplify_core/amplify_core.dart';

class OutboxStatusEvent extends DataStoreHubEventPayload {
  late bool isEmpty;
  OutboxStatusEvent(Map<dynamic, dynamic> serializedData) {
    isEmpty = serializedData['isEmpty'] as bool;
  }
}

File: packages/amplify_datastore/ios/Internal/AWSDataStorePlugin/Sync/Events/OutboxStatusEvent.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

/// Used as HubPayload for the `OutboxStatus`
public struct OutboxStatusEvent {
    /// status of outbox: true if there are no events in the outbox at the time the event was dispatched
    public let isEmpty: Bool

    public init(isEmpty: Bool) {
        self.isEmpty = isEmpty
    }
}
